dask_jobqueue/core.py

```diff
@@ -447,22 +453,47 @@ def __init__(
                 )
             )
 
+        if dashboard_address is not None:
+            raise ValueError(
+                "Please pass 'dashboard_address' through 'scheduler_options': use\n"
+                'cluster = {0}(..., scheduler_options={{"dashboard_address": ":12345"}}) rather than\n'
+                'cluster = {0}(..., dashboard_address="12435")'.format(
+                    self.__class__.__name__
+                )
+            )
+
+        if host is not None:
+            raise ValueError(
+                "Please pass 'host' through 'scheduler_options': use\n"
+                'cluster = {0}(..., scheduler_options={{"host": "your-host"}}) rather than\n'
+                'cluster = {0}(..., host="your-host")'.format(self.__class__.__name__)
+            )
+
         default_config_name = self.job_cls.default_config_name()
         if config_name is None:
             config_name = default_config_name
 
         if interface is None:
             interface = dask.config.get("jobqueue.%s.interface" % config_name)
+        if scheduler_options is None:
+            scheduler_options = {}
+
+        default_scheduler_options = {
+            "protocol": protocol,
+            "dashboard_address": ":8787",
+            "security": security,
+        }
+        # scheduler_options overrides parameters common to both workers and scheduler
+        scheduler_options = dict(default_scheduler_options, **scheduler_options)
+
+        # Use the same network interface as the workers if scheduler ip has not
+        # been set through scheduler_options via 'host' or 'interface'
+        if "host" not in scheduler_options and "interface" not in scheduler_options:
+            scheduler_options["interface"] = interface
```

Comment:
Should'nt we just put interface in the `default_scheduler_options`? Won't it be overriden if defined in the `scheduler_options`?

Or is this because of a problem in distributed if host and interface do not match?

Reply:
I should probably test this edge case indeed. My line of reasoning is like this (as I tried to explain in the comment):
- `interface` parameter applies primarily to workers. It should apply to the scheduler only if the scheduler IP is not already set by either `host` or `interface` in the `scheduler_options` dict.